R/filter_exist.R

```diff
@@ -0,0 +1,176 @@
+#' Returns records that fit into existing by groups in a filtered source dataset
+#'
+#' Returns all records that fit into an existing by group in a filtered source dataset.
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified in the `by_vars` parameter are expected in this dataset.
+#'
+#' @param dataset_add Source dataset
+#'
+#'   The source dataset, which determines the by groups returned in the input dataset,
+#'   based on the groups that exist in this dataset after being subset by `filter_add`.
+#'
+#'   The variables specified in the `by_vars` and `filter_add` parameters are expected
+#'   in this dataset.
+#'
+#' @param by_vars Grouping variables
+#'
+#'   A list of variable names specified within `exprs()` is expected.
+#'
+#' @param filter_add Filter for the source dataset
+#'
+#'   The filter condition which will be used to subset the source dataset.
+#'   Alternatively, if no filter condition is supplied, no subsetting of the source
+#'   dataset will be performed.
+#'
+#'   Default: `TRUE` (i.e. no filtering will be performed)
+#'
+#' @details Returns the records in `dataset` which match an existing by group in `dataset_add`,
+#'   after being filtered according to `filter_add`. If there are no by groups that exist
+#'   in both datasets, an empty dataset will be returned.
+#'
+#' @return The records in the input dataset which are contained within an existing by group in
+#'   the filtered source dataset.
+#'
+#' @keywords utils_fil
+#'
+#' @family utils_fil
+#'
+#' @export
+#'
+#' @examples
+#' # Get demographic information about subjects who have suffered from moderate or
+#' # severe fatigue
+#'
+#' library(tibble)
+#'
+#' adsl <- tribble(
+#'   ~USUBJID,      ~AGE, ~SEX,
+#'   "01-701-1015", 63,   "F",
+#'   "01-701-1034", 77,   "F",
+#'   "01-701-1115", 84,   "M",
+#'   "01-701-1146", 75,   "F",
+#'   "01-701-1444", 63,   "M"
+#' )
+#'
+#' adae <- tribble(
+#'   ~USUBJID,      ~AEDECOD,                    ~AESEV,     ~AESTDTC,
+#'   "01-701-1015", "DIARRHOEA",                 "MODERATE", "2014-01-09",
+#'   "01-701-1034", "FATIGUE",                   "SEVERE",   "2014-11-02",
+#'   "01-701-1034", "APPLICATION SITE PRURITUS", "MODERATE", "2014-08-27",
+#'   "01-701-1115", "FATIGUE",                   "MILD",     "2013-01-14",
+#'   "01-701-1146", "FATIGUE",                   "MODERATE", "2013-06-03"
+#' )
+#'
+#' filter_exist(
+#'   dataset = adsl,
+#'   dataset_add = adae,
+#'   by_vars = exprs(USUBJID),
+#'   filter_add = AEDECOD == "FATIGUE" & AESEV %in% c("MODERATE", "SEVERE")
+#' )
+#'
+filter_exist <- function(
+    dataset,
+    dataset_add,
+    by_vars,
+    filter_add = TRUE) {
```

Comment:
Usually we use `NULL` as default value. However, using `TRUE` does not require `filter_if()`.

@bms63 , @zdz2101 , what do you think? Should we change our approach?

Reply:
Hi @bms63, @zdz2101, @bundfussr,
Just wondering if there's any further thoughts on this?

Reply:
I think if we do change it from `NULL` to `TRUE` we should do it for all function arguments, so the merged/joined generics would be affected, perhaps throw it as a separate issue/PR for future?

Reply:
@bundfussr @bms63 @zdz2101, I have created issue #1911 for this discussion. I will update this PR to use our current approach of using `NULL` so that these functions will be ready for the upcoming release. Then, we can make a decision on this separately.